### PR TITLE
Code cleanup in actionlog plugin

### PR DESCRIPTION
**Pull request by HLeithner**
Pull Request for Issue #31378 .

### Summary of Changes
* Updated array() to [] syntax
* removed strtoupper on language hasKey call
* replaced deprecated functions `JFactory::getUser()` and `JFactory::getUser($userId)`

### Testing Instructions
Activate actionlog

### Actual result BEFORE applying this Pull Request
Everything works

### Expected result AFTER applying this Pull Request
Still works

### Documentation Changes Required
No

### CODE REVIEW

**Comment by infograf768** [comment]
drone not happy => phpcs

**Comment by ceford** [comment]
I just migrated to PHP 8 - logged in via different browser to create log events. Works OK. On logout I get this:

Joomla\CMS\Application\WebApplication::loadIdentity(): Argument #1 ($identity) must be of type ?Joomla\CMS\User\User, int given, called in /Users/ceford/Sites/j4test/plugins/actionlog/joomla/joomla.php on line 894

Don't know what to make of this!
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33072">issues.joomla.org/tracker/joomla-cms/33072</a>.</sub>

**Comment by joomdonation** [comment]
Personal, I would not include the changes from long array syntax to short array syntax on this PR to avoid potential merge conflicts when merge changes from staging to 4.0-dev. When we really want to make this kind of change, there is tool to allow us to make change through our whole codebase. Also, there is error as reported by @ceford . Could you please look at it?

File: plugins/actionlog/joomla/joomla.php [hunk 1]
```diff
@@ -891,7 +891,7 @@ public function onUserLogout($user, $options = array())
 			return;
 		}
 
-		$loggedOutUser = User::getInstance($user['id']);
+		$loggedOutUser = $this->app->loadIdentity($user['id']);
```
Comment (joomdonation) [inline plugins/actionlog/joomla/joomla.php:896]: This is the line causing the error. How about change it to:

```php
$loggedOutUser = $this->app->getContainer()->get('user.factory')->loadUserById($user['id']);
```

**Comment by sandewt** [comment]
https://github.com/joomla/joomla-cms/blob/3aae81918b1e06e4e1f0377b59413428122e670e/plugins/actionlog/joomla/joomla.php#L1162

Should be:

`$oldVersion = Text::_('JLIB_UNKNOWN'); `

and add:

`use Joomla\CMS\Language\Text;`

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 3aae81918b1e06e4e1f0377b59413428122e670e<br /><br />Tested successfully in Beta8-dev of 16 April using Wampserver 3.2.4 and PHP 8.0.2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33072">issues.joomla.org/tracker/joomla-cms/33072</a>.</sub>

**Comment by ceford** [comment]
I have been doing other stuff so catching up with testing. Working through some of the actions that invoke the actions log plugin, after front end logout I got an error:

     0 Joomla\CMS\Application\WebApplication::loadIdentity(): Argument #1 ($identity) must be of type ?Joomla\CMS\User\User, int given, called in /Users/ceford/Sites/j4test/plugins/actionlog/joomla/joomla.php on line 894 

Stopped testing there. This is using PHP 8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33072">issues.joomla.org/tracker/joomla-cms/33072</a>.</sub>

**Comment by PhilETaylor** [comment]
> Updated array() to [] syntax

Currently this is against the Joomla Coding Guidelines - please merge https://github.com/joomla/coding-standards/pull/273 ;-)

File: plugins/actionlog/joomla/joomla.php [hunk 2]
```diff
@@ -1066,14 +1070,14 @@ public function onAfterLogExport($group = '')
 	public function onAfterPurge($group = 'all')
 	{
 		$context = $this->app->input->get('option');
-		$user    = Factory::getUser();
+		$user    = $this->app->getIdentity();;
```
Comment (sandewt) [inline plugins/actionlog/joomla/joomla.php:1075]: One ';' to much:

'$this->app->getIdentity();;'

should  be

`$this->app->getIdentity();`

**Comment by chmst** [comment]
@HLeithner could you resolve the conflict, please?

**Comment by HLeithner** [comment]
This pull request has automatically rebased to 4.2-dev.

**Pull request closed without merging**